File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/ConnectionModel.java
package io.qimia.uhrwerk.common.model;

import com.google.common.base.Objects;
import java.io.Serializable;

public class ConnectionModel extends StateModel implements Serializable {

  private static final long serialVersionUID = -3946000637155224648L;

  String name;
  ConnectionType type;
  String path;
  String jdbcUrl;
  String jdbcDriver;
  String jdbcUser;
  String jdbcPass;
  String awsAccessKeyID;
  String awsSecretAccessKey;

  public static ConnectionModelBuilder builder() {
    return new ConnectionModelBuilder();
  }

  public ConnectionModel(ConnectionModelBuilder builder) {
    super(builder.deactivatedTs);
    this.id = builder.id;
    this.name = builder.name;
    this.type = builder.type;
    this.path = builder.path;
    this.jdbcUrl = builder.jdbcUrl;
    this.jdbcDriver = builder.jdbcDriver;
    this.jdbcUser = builder.jdbcUser;
    this.jdbcPass = builder.jdbcPass;
    this.awsAccessKeyID = builder.awsAccessKeyID;
    this.awsSecretAccessKey = builder.awsSecretAccessKey;
  }

  public String getName() {
    return name;
  }

  public ConnectionType getType() {
    return type;
  }

  public void setType(ConnectionType type) {
    this.type = type;
  }

  public String getJdbcUrl() {
    return jdbcUrl;
  }

  public void setJdbcUrl(String jdbcUrl) {
    this.jdbcUrl = jdbcUrl;
  }

  public String getJdbcDriver() {
    return jdbcDriver;
  }

  public void setJdbcDriver(String jdbcDriver) {
    this.jdbcDriver = jdbcDriver;
  }

  public String getJdbcUser() {
    return jdbcUser;
  }

  public void setJdbcUser(String jdbcUser) {
    this.jdbcUser = jdbcUser;
  }

  public String getJdbcPass() {
    return jdbcPass;
  }

  public void setJdbcPass(String jdbcPass) {
    this.jdbcPass = jdbcPass;
  }

  public String getAwsAccessKeyID() {
    return awsAccessKeyID;
  }

  public void setAwsAccessKeyID(String awsAccessKeyID) {
    this.awsAccessKeyID = awsAccessKeyID;
  }

  public String getAwsSecretAccessKey() {
    return awsSecretAccessKey;
  }

  public void setAwsSecretAccessKey(String awsSecretAccessKey) {
    this.awsSecretAccessKey = awsSecretAccessKey;
  }

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof ConnectionModel)) {
      return false;
    }
    if (!super.equals(o)) {
      return false;
    }
    ConnectionModel that = (ConnectionModel) o;
    return Objects.equal(getName(), that.getName())
        && getType() == that.getType()
        && Objects.equal(getPath(), that.getPath())
        && Objects.equal(getJdbcUrl(), that.getJdbcUrl())
        && Objects.equal(getJdbcDriver(), that.getJdbcDriver())
        && Objects.equal(getJdbcUser(), that.getJdbcUser())
        && Objects.equal(getJdbcPass(), that.getJdbcPass())
        && Objects.equal(getAwsAccessKeyID(), that.getAwsAccessKeyID())
        && Objects.equal(getAwsSecretAccessKey(), that.getAwsSecretAccessKey());
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(
        super.hashCode(),
        getName(),
        getType(),
        getPath(),
        getJdbcUrl(),
        getJdbcDriver(),
        getJdbcUser(),
        getJdbcPass(),
        getAwsAccessKeyID(),
        getAwsSecretAccessKey());
  }

  @Override
  public String toString() {
    return "ConnectionModel{"
        + "name='"
        + name
        + '\''
        + ", type="
        + type
        + ", path='"
        + path
        + '\''
        + ", jdbcUrl='"
        + jdbcUrl
        + '\''
        + ", jdbcDriver='"
        + jdbcDriver
        + '\''
        + ", jdbcUser='"
        + jdbcUser
        + '\''
        + ", jdbcPass='"
        + jdbcPass
        + '\''
        + ", awsAccessKeyID='"
        + awsAccessKeyID
        + '\''
        + ", awsSecretAccessKey='"
        + awsSecretAccessKey
        + '\''
        + '}';
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/model/ConnectionModel.kt
package io.qimia.uhrwerk.common.metastore.model

import java.sql.Timestamp
import java.time.LocalDateTime

data class ConnectionModel(
    var id: Long? = null,
    var name: String? = null,
    var type: ConnectionType? = null,
    var path: String? = null,
    var jdbcUrl: String? = null,
    var jdbcDriver: String? = null,
    var jdbcUser: String? = null,
    var jdbcPass: String? = null,
    var awsAccessKeyID: String? = null,
    var awsSecretAccessKey: String? = null,
    var redshiftFormat: String? = null,
    var redshiftAwsIamRole: String? = null,
    var redshiftTempDir: String? = null,
    var description: String? = null,
    var hashKey: Long? = null,
    var deactivatedTs: LocalDateTime? = null,
    var createdTs: Timestamp? = null,
    var updatedTs: Timestamp? = null
) : BaseModel {
    override fun id(id: Long?) {
        this.id = id
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ConnectionModel) return false
        if (id != null && other.id != null) if (id != other.id) return false
        if (name != other.name) return false
        if (type != other.type) return false
        if (path != other.path) return false
        if (jdbcUrl != other.jdbcUrl) return false
        if (jdbcDriver != other.jdbcDriver) return false
        if (jdbcUser != other.jdbcUser) return false
        if (jdbcPass != other.jdbcPass) return false
        if (awsAccessKeyID != other.awsAccessKeyID) return false
        if (awsSecretAccessKey != other.awsSecretAccessKey) return false
        if (redshiftFormat != other.redshiftFormat) return false
        if (redshiftAwsIamRole != other.redshiftAwsIamRole) return false
        if (redshiftTempDir != other.redshiftTempDir) return false
        if (hashKey != null && other.hashKey != null) if (hashKey != other.hashKey) return false
        if (description != other.description) return false
        if (deactivatedTs != other.deactivatedTs) return false
        if (createdTs != other.createdTs) return false
        if (updatedTs != other.updatedTs) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        result = 31 * result + (type?.hashCode() ?: 0)
        result = 31 * result + (path?.hashCode() ?: 0)
        result = 31 * result + (jdbcUrl?.hashCode() ?: 0)
        result = 31 * result + (jdbcDriver?.hashCode() ?: 0)
        result = 31 * result + (jdbcUser?.hashCode() ?: 0)
        result = 31 * result + (jdbcPass?.hashCode() ?: 0)
        result = 31 * result + (awsAccessKeyID?.hashCode() ?: 0)
        result = 31 * result + (awsSecretAccessKey?.hashCode() ?: 0)
        result = 31 * result + (redshiftFormat?.hashCode() ?: 0)
        result = 31 * result + (redshiftAwsIamRole?.hashCode() ?: 0)
        result = 31 * result + (redshiftTempDir?.hashCode() ?: 0)
        result = 31 * result + (hashKey?.hashCode() ?: 0)
        result = 31 * result + (description?.hashCode() ?: 0)
        result = 31 * result + (deactivatedTs?.hashCode() ?: 0)
        result = 31 * result + (createdTs?.hashCode() ?: 0)
        result = 31 * result + (updatedTs?.hashCode() ?: 0)
        return result
    }


}